File: robocode2-bot-api-java/src/main/java/net/robocode2/events/GameStartedEvent.java
package net.robocode2.events;

import lombok.Builder;
import lombok.ToString;
import lombok.Value;
import net.robocode2.GameSetup;

/** Event occurring when game has just started */
@Value
@ToString
public class GameStartedEvent implements Message {
  /** The ID used for identifying your bot in the current battle */
  int myId;
  /** The game setup for the battle just started */
  GameSetup gameSetup;

  @Builder
  private GameStartedEvent(int myId, GameSetup gameSetup) {
    this.myId = myId;
    this.gameSetup = gameSetup;
  }
}

File: bot-api/python/src/robocode_tank_royale/bot_api/events/game_started_event.py
from dataclasses import dataclass

from ..game_setup import GameSetup
from ..initial_position import InitialPosition
from .event_abc import EventABC


@dataclass(frozen=True, repr=True)
class GameStartedEvent(EventABC):
    """
    Represents an event triggered when the game starts.

    Attributes:
        my_id (int): The unique identifier for your bot in the current battle.
        initial_position (InitialPosition): The starting position of the bot.
        game_setup (GameSetup): The configuration details for the game that has just started.
    """

    my_id: int
    initial_position: InitialPosition
    game_setup: GameSetup
